rclpy/test/test_task.py

```diff
@@ -322,6 +322,34 @@ def cb(fut):
         f.add_done_callback(cb)
         assert called
 
+    def test_set_result_on_done_future_without_exception(self) -> None:
+        f = Future()
+        f.set_result(None)
+        self.assertTrue(f.done())
```

Comment:
To get coverage we should check the negatives too. 
```suggestion
        self.assertTrue(f.done())
        self.assertFalse(f.cancelled())
```